### PR TITLE
feat: add kubeflow-dashboard sidebar relation

### PR DESCRIPTION
**Pull request by ca-scribner**
Adds the sidebar relation for kubeflow-dashboard, plus the library to manage it.  This allows this charm to request a link in the kubeflow dashboard's sidebar.

### CODE REVIEW

**Comment by i-chvets** [comment]
I would add integration tests for this new relation.

**Comment by ca-scribner** [comment]
After discussion within the team, we decided to reduce the testing here. Each application using the kubeflow-dashboard-links lib will have no integration tests for that in the charm itself, and instead we will catch that at bundle-level tests (captured as issue https://github.com/canonical/bundle-kubeflow/issues/636)

File: charms/kfp-ui/src/charm.py [hunk 1]
```diff
@@ -47,6 +51,54 @@ def __init__(self, *args):
...
+                    desc="Pipelines",
+                    link="/pipeline/",
+                    location="quick",
+                ),
+                DashboardLink(
+                    text="View all pipeline runs",
+                    desc="Pipelines",
+                    link="/pipeline/#/runs",
+                    location="quick",
+                )
```
Comment (ca-scribner) [inline charms/kfp-ui/src/charm.py:98]:
```suggestion
                ),
```

**Comment by ca-scribner** [comment]
tests are failing but I think unrelated to the new dashboard links in this pr

**Pull request merged**